at_client/lib/src/client/at_client_impl.dart

```diff
@@ -308,19 +325,68 @@ class AtClientImpl implements AtClient {
   @override
   Future<bool> put(AtKey atKey, dynamic value,
       {bool isDedicated = false}) async {
-    // Perform atKey validations.
-    await AtClientValidation.validateAtKey(atKey);
+    AtResponse atResponse = AtResponse();
+    // If the value is neither String nor List<int> throw exception
+    if (value is! String && value is! List<int>) {
+      // TODO: Throw AtValueException
+      throw AtClientException('AT014',
+          'Invalid value type found ${value.runtimeType}. Expected String or List<int>');
+    }
+    if (value is String) {
+      atResponse = await putText(atKey, value);
+    }
+    if (value is List<int>) {
+      atResponse = await putBinary(atKey, value);
+    }
+    return atResponse.response.isNotEmpty;
+  }
+
+  /// put's the text data into the keystore
+  @override
+  Future<AtResponse> putText(AtKey atKey, String value) async {
+    // Set the default metadata if not already set.
+    atKey.metadata ??= Metadata();
+    // Setting metadata.isBinary to false for putText
+    atKey.metadata!.isBinary = false;
+    return await _putInternal(atKey, value);
+  }
+
+  /// put's the binary data(e.g. images, files etc) into the keystore
+  @override
+  Future<AtResponse> putBinary(AtKey atKey, List<int> value) async {
+    // Set the default metadata if not already set.
+    atKey.metadata ??= Metadata();
+    // Setting metadata.isBinary to true for putBinary
+    atKey.metadata!.isBinary = true;
+    // Base2e15.encode method converts the List<int> type to String.
+    return await _putInternal(atKey, Base2e15.encode(value));
+  }
+
+  Future<AtResponse> _putInternal(AtKey atKey, dynamic value) async {
+    // Performs the put request validations.
+    AtClientValidation.validatePutRequest(atKey, value, preference!);
     // Set sharedBy to currentAtSign if not set.
     if (atKey.sharedBy == null || atKey.sharedBy!.isEmpty) {
```

Comment:
we can have a private method to check a string is null or empty..we have this check in multiple places

Reply:
Implemented isNull method to verify if a field is null or empty